### DISCUSSION TITLE
Not able to resign my IPA file. Can someone please look into it quickly give a solution?

### DISCUSSION THREAD

**Discussion by SubhamDyno** [fastlane/fastlane#19176] [Q&A]
<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 11.5                                                    |
| Ruby                        | 3.0.2                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.30.1 (Apple Git-130)                      |
| Installation Source         | /usr/local/Cellar/fastlane/2.189.0/libexec/bin/fastlane |
| Host                        | macOS 11.5 (20G71)                                      |
| Ruby Lib Dir                | /usr/local/Cellar/ruby/3.0.2/lib                        |
| OpenSSL Version             | OpenSSL 1.1.1k  25 Mar 2021                             |
| Is contained                | false                                                   |
| Is homebrew                 | true                                                    |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 12.5.1                                                  |
| Swift Version               | 5.4.2                                                   |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.189.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.5.0        |
| atomos                          | 0.1.3        |
| CFPropertyList                  | 3.0.3        |
| claide                          | 1.0.3        |
| colored2                        | 3.1.2        |
| nanaimo                         | 0.3.0        |
| rexml                           | 3.2.5        |
| xcodeproj                       | 1.20.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.7.0        |
| terminal-table                  | 1.8.0        |
| plist                           | 3.6.0        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.8.0        |
| multipart-post                  | 2.0.0        |
| word_wrap                       | 1.0.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| artifactory                     | 3.0.15       |
| babosa                          | 1.0.4        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| excon                           | 0.85.0       |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-httpclient              | 1.0.1        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.2.0        |
| faraday-patron                  | 1.0.0        |
| ruby2_keywords                  | 0.0.5        |
| faraday                         | 1.5.1        |
| unf_ext                         | 0.0.7.7      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| http-cookie                     | 1.0.4        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.0.0        |
| fastimage                       | 2.2.4        |
| gh_inspector                    | 1.1.3        |
| json                            | 2.5.1        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| rubyzip                         | 2.3.2        |
| security                        | 0.1.3        |
| xcpretty-travis-formatter       | 1.0.1        |
| dotenv                          | 2.7.6        |
| bundler                         | 2.2.22       |
| simctl                          | 1.6.8        |
| jwt                             | 2.2.3        |
| uber                            | 0.1.0        |
| declarative                     | 0.0.20       |
| trailblazer-option              | 0.1.1        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| mini_mime                       | 1.1.0        |
| memoist                         | 0.16.2       |
| multi_json                      | 1.15.0       |
| os                              | 1.1.1        |
| signet                          | 0.15.0       |
| googleauth                      | 0.16.2       |
| httpclient                      | 2.8.3        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.4.1        |
| google-apis-playcustomapp_v1    | 0.5.0        |
| google-apis-androidpublisher_v3 | 0.9.0        |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.1.0        |
| google-cloud-core               | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.6.0        |
| google-apis-storage_v1          | 0.6.0        |
| rake                            | 13.0.6       |
| digest-crc                      | 0.6.4        |
| google-cloud-storage            | 1.34.1       |
| emoji_regex                     | 3.2.2        |
| jmespath                        | 1.4.0        |
| aws-partitions                  | 1.480.0      |
| aws-eventstream                 | 1.1.1        |
| aws-sigv4                       | 1.2.4        |
| aws-sdk-core                    | 3.117.0      |
| aws-sdk-kms                     | 1.44.0       |
| aws-sdk-s3                      | 1.96.2       |
| tsort                           | 0.1.0        |
| uri                             | 0.10.1       |
| set                             | 1.0.1        |
| forwardable                     | 1.3.2        |
| logger                          | 1.4.3        |
| pathname                        | 0.1.0        |
| shellwords                      | 0.1.0        |
| cgi                             | 0.2.0        |
| date                            | 3.1.0        |
| timeout                         | 0.1.1        |
| stringio                        | 3.0.0        |
| openssl                         | 2.2.0        |
| io-nonblock                     | 0.1.0        |
| ipaddr                          | 1.2.2        |
| io-wait                         | 0.1.0        |
| zlib                            | 1.1.0        |
| resolv                          | 0.2.0        |
| securerandom                    | 0.1.0        |
| digest                          | 3.0.0        |
| time                            | 0.1.0        |
| open-uri                        | 0.1.0        |
| mutex_m                         | 0.1.1        |
| net-protocol                    | 0.1.0        |
| ostruct                         | 0.3.1        |
| english                         | 0.7.1        |
| erb                             | 2.2.0        |
| strscan                         | 3.0.0        |
| optparse                        | 0.1.0        |
| abbrev                          | 0.1.0        |
| io-console                      | 0.5.7        |
| tempfile                        | 0.1.1        |
| delegate                        | 0.2.0        |
| fileutils                       | 1.5.0        |
| tmpdir                          | 0.1.2        |
| base64                          | 0.1.0        |
| singleton                       | 0.1.1        |
| net-http                        | 0.1.1        |
| open3                           | 0.1.1        |
| nkf                             | 0.1.0        |
| prettyprint                     | 0.1.0        |
| pp                              | 0.1.0        |
| find                            | 0.1.0        |
| yaml                            | 0.1.1        |
| psych                           | 3.3.0        |
</details>

*generated on:* **2021-07-29**
</details>

**Comment by crazymanish** [comment]
Hi @SubhamDyno 

we need a couple of details from you to debug the issue:

- what was the command used, which is failing? 
- can you share the stack trace/ logs? 
- was it working earlier and stop working suddenly? if yes, what was the last working version?

thanks!

**Comment by SubhamDyno** [comment]
Hello Manish

1. Command that was used -
`fastlane sigh resign -i --verbose "iPhone Distribution: xxx Inc." -p /Users/subhampatra/Desktop/test/dir/xxx.mobileprovision /Users/subhampatra/Desktop/test/dir/com.xxx.xxx.ipa use_app_entitlements`

2. Please find the trace Logs here - 
[logs.txt](https://github.com/fastlane/fastlane/files/6905107/logs.txt)

3. I am trying it for the very first-time.

Another note to be taken - I am able to sign the application in manual process without Fastlane process bypassing few warnings.
But with fastlane attached is the issue.

**Comment by crazymanish** [comment]
Hi,  

That's is not an issue then, Quickly checked your logs, **_i think_** you are using the wrong command...
moving it discussion so Fastlane community can help us

**Reply by RomanPoschl** [reply]
Hi,
I checked @SubhamDyno logs and I am gettings same error when trying to resign my IPA in Azure DevOps. But when Fastlane is trying to sign libSkiaSharp.framework

**Comment by Richard7119** [comment]
The answers posted here all didn't quite work for me. They mainly skipped signing embedded frameworks (or including the entitlements).

Here's what's worked for me (it assumes that one ipa file exists is in the current directory):

PROVISION="/path/to/file.mobileprovision"
CERTIFICATE="Name of certificate: To sign with" # must be in the keychain

unzip -q *.ipa [jcpenneykiosk](https://www.jcpenneykiosk.bid/)
rm -rf Payload/*.app/_CodeSignature/

# Replace embedded provisioning profile
cp "$PROVISION" Payload/*.app/embedded.mobileprovision

# Extract entitlements from app
codesign -d --entitlements :entitlements.plist Payload/*.app/

# Re-sign embedded frameworks
codesign -f -s "$CERTIFICATE" --entitlements entitlements.plist Payload/*.app/Frameworks/*

# Re-sign the app (with entitlements)
codesign -f -s "$CERTIFICATE" --entitlements entitlements.plist Payload/*.app/

zip -qr resigned.ipa Payload

# Cleanup
rm entitlements.plist
rm -r Payload/
